File: backend/app/models/daily_checkin.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, Integer, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.db.database import Base


class DailyCheckIn(Base):
    __tablename__ = "daily_checkins"

    __table_args__ = (
        UniqueConstraint("user_id", "checkin_date", name="uq_user_checkin_date"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, index=True)

    user_id: Mapped[int] = mapped_column(
        ForeignKey("users.id"),
        nullable=False,
        index=True,
    )

    checkin_date: Mapped[date] = mapped_column(Date, nullable=False, index=True)

    mood_score: Mapped[int | None] = mapped_column(Integer, nullable=True)
    energy_score: Mapped[int | None] = mapped_column(Integer, nullable=True)
    focus_score: Mapped[int | None] = mapped_column(Integer, nullable=True)
    stress_score: Mapped[int | None] = mapped_column(Integer, nullable=True)
    sleep_quality_score: Mapped[int | None] = mapped_column(Integer, nullable=True)

    note: Mapped[str | None] = mapped_column(String(1500), nullable=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
        nullable=False,
    )

File: frontend/src/types/dailyCheckin.ts
export type DailyCheckInCreate = {
  checkin_date: string;
  mood_score: number | null;
  energy_score: number | null;
  focus_score: number | null;
  stress_score: number | null;
  sleep_quality_score: number | null;
  note: string | null;
};

export type DailyCheckInResponse = {
  id: number;
  user_id: number;

  checkin_date: string;

  mood_score: number | null;
  energy_score: number | null;
  focus_score: number | null;
  stress_score: number | null;
  sleep_quality_score: number | null;

  note: string | null;

  created_at: string;
  updated_at: string;
};